crates/loam-cli/src/commands/build/build_clients.rs

```diff
@@ -0,0 +1,229 @@
+#![allow(clippy::struct_excessive_bools)]
+use crate::commands::build::env_toml;
+use clap::Parser;
+use soroban_cli::commands::NetworkRunnable;
+use soroban_cli::{commands as cli, CommandParser};
+use std::collections::BTreeMap as Map;
+use std::fmt::Debug;
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, clap::ValueEnum)]
+pub enum LoamEnv {
+    Development,
+    Testing,
+    Staging,
+    Production,
+}
+
+#[derive(Parser, Debug, Clone)]
+pub struct Cmd {
+    #[arg(long, default_value = ".")]
+    pub workspace_root: std::path::PathBuf,
+    #[arg(env = "LOAM_ENV", value_enum, default_value = "production")]
+    pub env: LoamEnv,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    EnvironmentsToml(#[from] env_toml::Error),
+    #[error("⛔ ️invalid network: must either specify a network name or both network_passphrase and rpc_url")]
+    MalformedNetwork,
+    #[error(transparent)]
+    ParsingNetwork(#[from] cli::network::Error),
+    #[error(transparent)]
+    GeneratingKey(#[from] cli::keys::generate::Error),
+    #[error("⛔ ️can only have one default account; marked as default: {0:?}")]
+    OnlyOneDefaultAccount(Vec<String>),
+    #[error("⛔ ️you need to provide at least one account, to use as the source account for contract deployment and other operations")]
+    NeedAtLeastOneAccount,
+    #[error("⛔ ️No contract named {0:?}")]
+    BadContractName(String),
+    #[error(transparent)]
+    ContractInstall(#[from] cli::contract::install::Error),
+    #[error(transparent)]
+    ContractDeploy(#[from] cli::contract::deploy::wasm::Error),
+    #[error(transparent)]
+    ContractBindings(#[from] cli::contract::bindings::typescript::Error),
+}
+
+impl Cmd {
+    pub async fn run(&self) -> Result<(), Error> {
+        let current_env = env_toml::Environment::get(&self.workspace_root, self.loam_env())?;
+
+        if current_env.is_none() {
+            return Ok(());
+        }
+        let current_env = current_env.unwrap();
+
+        self.add_network_to_env(&current_env.network)?;
+        self.handle_accounts(&current_env.accounts).await?;
+        self.handle_contracts(&current_env.contracts).await?;
+
+        Ok(())
+    }
+
+    fn loam_env(&self) -> String {
+        format!("{:?}", self.env).to_lowercase()
+    }
+
+    /// Parse the network settings from the environments.toml file and set STELLAR_RPC_URL and
+    /// STELLAR_NETWORK_PASSPHRASE.
+    ///
+    /// We could set STELLAR_NETWORK instead, but when importing contracts, we want to hard-code
+    /// the network passphrase. So if given a network name, we use soroban-cli to fetch the RPC url
+    /// & passphrase for that named network, and still set the environment variables.
+    fn add_network_to_env(&self, network: &env_toml::Network) -> Result<(), Error> {
+        let rpc_url = &network.rpc_url;
+        let network_passphrase = &network.network_passphrase;
```

Comment:
You could do a match here on `&network`. It would  be a bit cleaner, especially the nested if lets. e.g.

```rust
match network {
    Network { name: Some(name), ..}  => ...
    Network { rpc_url: Some(rpc_url), network_passphrase: Some(network_passphrase, ..} => ...
    _ => return Err
}
```